src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/BookerPrisonerInfoClient.kt

```diff
@@ -28,39 +26,40 @@ class BookerPrisonerInfoClient(
   }
 
   fun getPermittedPrisonerInfo(
-    offenderSearchPrisoner: PrisonerDto,
     bookerPrisoner: PermittedPrisonerForBookerDto,
   ): BookerPrisonerInfoDto? {
-    val prisonCode = offenderSearchPrisoner.prisonId!!
-    val prisonMono = visitSchedulerClient.getPrisonAsMono(prisonCode)
-    val visitBalancesDtoMono = prisonApiClient.getVisitBalancesAsMono(offenderSearchPrisoner.prisonerNumber)
-
-    Mono.zip(prisonMono, visitBalancesDtoMono).block(apiTimeout).also {
-      val prison = it?.t1?.getOrNull()
+    val prisonCode = bookerPrisoner.prisonCode
+    val prisonerId = bookerPrisoner.prisonerId
+    val offenderSearchPrisonerDtoMono = prisonerSearchClient.getPrisonerByIdAsMonoEmptyIfNotFound(prisonerId)
```

Comment:
earlier we used to do the offenderSearch call prior to calling this method but now that we get the prison code from Prison Registry we are doing an offender search in parallel to the other calls.